.github/workflows/bioblend.yaml

```diff
@@ -0,0 +1,128 @@
+name: Tests
+on:
+  pull_request:
+    paths:
+      - "lib/galaxy/webapps/galaxy/api/**"
+  workflow_dispatch:
+
+concurrency:
+  group: test-${{ github.ref }}
+  cancel-in-progress: true
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    services:
+      postgres:
+        image: postgres
+        # Provide the password for postgres
+        env:
+          POSTGRES_PASSWORD: postgres
+        # Set health checks to wait until postgres has started
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          - 5432:5432
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        tox_env: [py38]
+        galaxy_version:
+          - dev
+          - release_24.1
+          - release_24.0
+          - release_23.2
+          - release_23.1
+          - release_23.0
+          - release_22.05
+          - release_22.01
+          - release_21.09
+          - release_21.05
+          - release_21.01
+          - release_20.09
+          - release_20.05
+          - release_20.01
+          - release_19.09
+          - release_19.05
+        include:
+          - os: ubuntu-latest
+            tox_env: py313
+            galaxy_version: dev
+          # Cannot test on macOS because service containers are not supported
+          # yet: https://github.community/t/github-actions-services-available-on-others-vms/16916
+          # - os: macos-latest
+          #   tox_env: py38
+          #   galaxy_version: dev
```

Comment:
You don't need this big matrix. Drop `os` and `galaxy_version` (you should use `actions/checkout` to get galaxy at the commit being tested), and choose a Python version for Galaxy (I'd say '3.8') and one for BioBlend ('3.12') to test on.

Reply:
What about the `tox_env`, as it determines the BioBlend python version?

Reply:
(Amended the above comments to say Python 3.13 for BioBlend).